### DISCUSSION TITLE
Need some indication when the email is sent

### DISCUSSION THREAD

**Discussion by rafaelcn** [elementary/mail#549] [Ideas]
Currently there's no way to know unless checking in the app when the email was sent. It should be a nice feature to have a sound or a notification indicating that the email was sent to the mail server.

**Comment by danirabbit** [comment]
I think normally we assume success and notify on failure instead of the other way around

**Comment by dracid** [comment]
I second this. Having an actual notification gives _peace of mind_ that the message was SENT and not stuck in OUTGOING process (for whatever reason).
Thanks!

**Comment by rafaelcn** [comment]
@danrabbit that's not common sense when a user uses some application. The user wants feedback, let the users know when the status of something has changed/updated. Although this kind of mindset can easily bloat the interface with notifications and because of that this kind of feature should be well designed. Something like Apple does on the iPhone, let the user know with some silent bar on the app or, if notification not blocked, some very brief sound should be activated.

**Comment by marbetschar** [comment]
Right now the composer view is blocked while a message is sent. How about having the icon of the "Send" button replaced with a spinner which indicates the activity while we are sending and ultimately have a success check mark there as icon just before the composer gets closed?

Here's a rough sketch of what I'm thinking about:

![Screenshot from 2021-09-16 08-41-20](https://user-images.githubusercontent.com/392542/133563550-ab82e5eb-4107-40db-8953-a96a666f2c12.png)

**Reply by rafaelcn** [reply]
It is blocked while a message is sending or sent? This idea blocks/prevents the user to do anything else but to pay attention to that spinner and see if it actually sent the message. I think it needs to be more subtle than that. A sound indicator if the user decided to not mute the entire OS sound would be fine.

**Reply by marbetschar** [reply]
Not the app is blocked - just the composer view: Because while one is sending an email, we obviously no longer want nor should we interact with the composer (= editor). That's exactly how it works right now (as described above).

Also we don't want to send a bunch of different notifications around just for "confirmation", because the basic philosophy is quite the opposite: The user should be notified if something _goes wrong_. Otherwise they should assume it "just works". If it doesn't, then this clearly is a bug and needs to be properly fixed - not worked around with something like a notification.

**To conclude:** What I'm proposing here is a subtle change to build this trust. For the ones who do not trust messages to be sent successfully, we show a small indication while the message is "leaving the computer" and end this interaction with a success icon. Yes, this requires the skeptical ones to wait for the success icon the first few times - but I'll bet once trust is built, you probably won't wait for it any more. Also because we are animating between different states here, it won't "feel" like a long time - so even if you wait for the success icon every time, it shouldn't feel like "a huge blocker".

**Reply by rafaelcn** [reply]
I wasn't talking about using the eOS notification system. While I agree with that philosophy (a GNU philosophy you find all around the Linux userland (the shell)) I disagree on a user perspective of an application. About your solution, what if I want to write another email? Is the composer (editor) locked or does it prevents me from doing that? If it doesn't how can I see the notification that the mail was sent? 

Your solution would work in a way that outlook handles emails? It uses tabs on the bottom of the screen to let users quickly change context. If that context has the button notification that the email was sent it would work.

**Reply by marbetschar** [reply]
> About your solution, what if I want to write another email? Is the composer (editor) locked or does it prevents me from doing that? If it doesn't how can I see the notification that the mail was sent?

As described, my proposed solution does not alter the current behaviour of Mail - it just _adds_ something. That said, all of your questions here can easily be answered by _actually_ using the current version of Mail on elementary OS 6.

> Your solution would work in a way that outlook handles emails? It uses tabs on the bottom of the screen to let users quickly change context. If that context has the button notification that the email was sent it would work.

My previous sentence above already answers this question.

At this point I don't think we are really constructively getting anywhere. Don't get me wrong, I'm happy to help to make Mail better - but I rather prefer to invest my time in concrete problems.

To sum up: I just dropped by and provided a suggestion. If you don't like it thats fine by me 🤷
